server/issue.go

```diff
@@ -937,6 +915,54 @@ func getUpdateTask(database *api.Database, migrationType db.MigrationType, vcsPu
 	}, nil
 }
 
+// createTaskListForDatabaseCreate returns the task list of creating a database.
+func createTaskListForDatabaseCreate(instanceID int, payload api.TaskDatabaseCreatePayload, taskStatus api.TaskStatus) ([]api.TaskCreate, []api.TaskIndexDAG, error) {
+	var taskCreateList []api.TaskCreate
+	var taskIndexDAGList []api.TaskIndexDAG
```

Comment:
The restore task should come after create database. In this case, the taskIndexDAGList should be created after we know/create both tasks.